katdal/chunkstore_s3.py

```diff
@@ -113,10 +113,9 @@ def __getattr__(self, name):
     def read(self, size, *args, **kwargs):
         """Overload `read` method to detect truncated data source."""
         data = self._readable.read(size, *args, **kwargs)
-        bytes_read = len(data)
-        if bytes_read != size:
+        if data == b'' and size != 0:
             raise TruncatedRead('Error reading from S3 HTTP response: expected '
-                                '{} bytes, got {}'.format(size, bytes_read))
+                                '{} more byte(s), got EOF'.format(size))
```

Comment:
I'm having second thoughts about this change to only consider zero-length as truncation: from what I can tell, http.client.HTTPResponse will always block until it either gets EOF or the full requested amount, so a short read does indicate truncation. I haven't checked it for urllib3. But I guess let's first see what the new tests (to pause halfway) do.

Reply:
If I remember correct, `read` (as opposed to `readinto`) only gets used when reading the numpy header. I think that means the pause test might need to be extended to ensure that it pauses partway through the numpy header to fully test things.

Reply:
Done.